src/Core/Settings.h

```diff
@@ -88,6 +88,7 @@ class IColumn;
     M(Milliseconds, distributed_directory_monitor_max_sleep_time_ms, 30000, "Maximum sleep time for StorageDistributed DirectoryMonitors, it limits exponential growth too.", 0) \
     \
     M(Bool, distributed_directory_monitor_batch_inserts, false, "Should StorageDistributed DirectoryMonitors try to batch individual inserts into bigger ones.", 0) \
+    M(Bool, distributed_directory_monitor_split_batch_on_failure, true, "Should StorageDistributed DirectoryMonitors try to split batch into smaller in case of failures.", 0) \
```

Comment:
And since later I did the same splitting for corrupted batches I guess it is not a good idea to enable it by default, thoughts?